tarogarden/seedling.go

```diff
@@ -124,6 +129,47 @@ func (c Seedling) validateFields() error {
 	return nil
 }
 
+// validateGroupKey attempts to validate that the non-zero group key provided
+// with a seedling is owned by the daemon and can be used with this seedling.
+func (c Seedling) validateGroupKey(groups []*asset.AssetGroup) (
+	*asset.AssetGroup, error) {
+
+	// Scan for a group that matches our seedling group key.
+	var seedlingGroup *asset.AssetGroup
+	for i, group := range groups {
+		if bytes.Equal(group.GroupPubKey.SerializeCompressed(),
+			c.GroupKey.SerializeCompressed()) {
+
+			seedlingGroup = groups[i]
+		}
+	}
+
+	if seedlingGroup == nil {
+		return nil, fmt.Errorf("group key %v not found",
+			c.GroupKey.SerializeCompressed())
+	}
+
+	// We must be able to sign with the group key.
+	if !seedlingGroup.GroupKey.IsLocal() {
```

Comment:
When can the seedling group key be not local?

Reply:
When this is called, all we've verified about the group key is that it's a valid pubkey. Since we import group keys when receiving assets and importing proof files, we may have a foreign group key in the DB that the user could have provided. So we want to detect that early and not save this seedling.

Reply:
Aha, trying to mint an asset that you have received but did not issue 👍